### ISSUE TITLE
GapEncoder to support a sparse matrix (scipy) as input

### ISSUE THREAD

**Issue by ogencoglu** [skrub-data/skrub#1099]
### Problem Description

Currently it seems like GapEncoder is constrained to text as input. Would be great to be able to pass scipy sparse matrix and skip the vectorization thing it does.

### Feature Description

`.fit(X)` to support scipy sparse matrices for `X`

**Comment by GaelVaroquaux** [comment]
Thanks for your comment.
The GapEncoder is specifically made to encode strings and we will not repurpose it for other things as 1) some of the underlying code would not carry over, 2) it would not enable us to give a consistent user story and thus make the documentation confusing.

I think that what you are looking for is not the GapEncoder, but matrix factorization tools. For instance, the closest one in terms of statistical model is https://scikit-learn.org/stable/modules/generated/sklearn.decomposition.MiniBatchNMF.html . I do believe that it support's sparse matrices, but I'm not certain. You need to choose wisely the loss to have a factorization similar to the one we do in the GapEncoder.
Fun fact: the code of scikit-learn's MiniBatchNMF is actually evolved from the algorithmic code of the GapEncoder.

**Comment by ogencoglu** [comment]
Understood! Thanks for the swift reply.

**Comment by GaelVaroquaux** [comment]
> Understood! Thanks for the swift reply.

Thanks for understanding. I hope that you are successful with scikit-learn objects. Don't hesitate to tell us here. It won't change much for skrub, but it will make me happy.
